packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -336,7 +336,12 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
     protected forceUpdate({ resize }: TreeWidget.ForceUpdateOptions = { resize: true }): void {
         if (this.view && this.view.list) {
             if (resize && this.isVisible) {
-                this.view.cache.clearAll();
+                const rowIndex = this.getScrollToRow();
+                if (rowIndex !== undefined) {
+                    this.view.cache.clear(rowIndex, 0);
```

Comment:
it makes sense if it comes from `onSelectionChanged`, in all other cases it can cause bogus height

Reply:
try to resize the debug console for instance now:
<img width="475" alt="Screenshot 2020-07-10 at 08 44 52" src="https://user-images.githubusercontent.com/3082655/87124885-d3ec2c80-c289-11ea-903a-73cdf8ca2a76.png">